src/Promise.js

```diff
@@ -141,6 +144,22 @@ function Promise(handler, parent) {
     return me;
   };
 
+/**
+   * Calls a defined worker method with optional parameters
+   * @param {string} methodName Name of the method to be called as defined in the workerpool.worker() function
+   * @param {Array} [methodParameters] Array with optional parameters
+   */
+this.emit = function (methodName, methodParameters) {
+  if (this.worker?.worker && methodName) {
```

Comment:
Right now, an emitted event will be silently ignored when a worker is not yet attached (i.e. the task is queued). I think it would be better to throw an exception, or queue the message and execute. Or maybe return a boolean indicating whether the message was sent or not. What do you think?